samples/net/slip_throughput_client/src/main.c

```diff
@@ -0,0 +1,439 @@
+}
+
+/* Print final statistics */
+static void print_final_stats(void)
+{
+	uint32_t elapsed = k_uptime_get_32() - stats.start_time;
+	uint32_t elapsed_sec = elapsed / 1000;
+
+	if (elapsed_sec == 0) {
+		elapsed_sec = 1;
+	}
+
+	uint64_t tx_kbps = (stats.bytes_sent * 8) / (elapsed_sec * 1000);
+	uint64_t rx_kbps = (stats.bytes_received * 8) / (elapsed_sec * 1000);
+	uint32_t avg_rtt_us = stats.packets_received > 0 ?
+			      (uint32_t)(stats.total_rtt_us / stats.packets_received) : 0;
+
+	LOG_INF("");
+	LOG_INF("=== Final Statistics ===");
+	LOG_INF("Test duration: %u seconds", elapsed_sec);
+	LOG_INF("Packet size: %u bytes", test_params.packet_size);
+	LOG_INF("");
+	LOG_INF("TX: %llu packets, %llu bytes (%llu kbps)",
+		stats.packets_sent, stats.bytes_sent, tx_kbps);
+	LOG_INF("RX: %llu packets, %llu bytes (%llu kbps)",
+		stats.packets_received, stats.bytes_received, rx_kbps);
+	LOG_INF("");
+	LOG_INF("Packet loss: %llu (%.2f%%)",
+		stats.packets_sent - stats.packets_received,
+		stats.packets_sent > 0 ? 
+		(double)(stats.packets_sent - stats.packets_received) * 100.0 / stats.packets_sent : 0.0);
+	LOG_INF("Errors: %llu, Pattern errors: %llu, Timeouts: %llu",
+		stats.errors, stats.pattern_errors, stats.timeouts);
+	LOG_INF("");
+	LOG_INF("RTT (us): min=%u, avg=%u, max=%u",
+		stats.min_rtt_us, avg_rtt_us, stats.max_rtt_us);
+	LOG_INF("======================");
+}
+
+/* Run throughput test */
+static int run_test(void)
+{
+	int sock;
+	struct sockaddr_in server_addr;
+	uint8_t tx_buffer[MAX_PACKET_SIZE];
+	uint8_t rx_buffer[MAX_PACKET_SIZE];
+	int ret;
+	uint32_t seq = 0;
+	struct timeval timeout;
+	uint32_t test_end_time;
+	struct net_if *iface;
+
+	if (test_params.running) {
+		LOG_WRN("Test already running");
+		return -EBUSY;
+	}
+
+	if (test_params.packet_size > MAX_PACKET_SIZE) {
+		LOG_ERR("Packet size too large: %u (max: %u)",
+			test_params.packet_size, MAX_PACKET_SIZE);
+		return -EINVAL;
+	}
+
+	test_params.running = true;
+
+	LOG_INF("=== Starting Throughput Test ===");
+	LOG_INF("Server: %s:%u", test_params.server_ip, test_params.server_port);
+	LOG_INF("Packet size: %u bytes", test_params.packet_size);
+	LOG_INF("Duration: %u seconds", test_params.duration_sec);
+	LOG_INF("Delay: %u ms", test_params.delay_ms);
+
+	/* Find SLIP interface */
+	iface = net_if_get_first_by_type(&_net_l2_COBS_SERIAL);
+	if (!iface) {
+		LOG_ERR("No SLIP interface found");
+		test_params.running = false;
+		return -ENODEV;
+	}
+
+	/* Wait for interface to be up */
+	while (!net_if_is_up(iface)) {
+		LOG_INF("Waiting for SLIP interface to come up...");
+		k_sleep(K_MSEC(100));
+	}
+
+	/* Create UDP socket */
+	sock = zsock_socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	if (sock < 0) {
+		LOG_ERR("Failed to create socket: %d", errno);
+		test_params.running = false;
+		return -errno;
+	}
+
+	/* Set receive timeout */
+	timeout.tv_sec = 1;
+	timeout.tv_usec = 0;
+	ret = zsock_setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout));
```

Comment:
Please check the code and do not mix POSIX and network namespaced symbols. We recently merged a PR that adds a prefix to various network symbols in order to avoid conflicts with POSIX.
So here use `ZSOCK_SOL_SOCKET` and `ZSOCK_SO_RCVTIMEO`
If you want to check the usage of the correct APIs, you can set `CONFIG_NET_NAMESPACE_COMPAT_MODE=n` to disable the compatibility mode and make sure correct symbols are used.

Reply:
These samples will be cleaned up as I mentioned in the note in PR description.